packages/core/admin/ee/server/services/audit-logs/process-event.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+module.exports = function processEvent(strapi, name, ...args) {
+  const defaultGetPayload = (...args) => args[0];
+  const getAuthPayload = (...args) => ({
+    user: strapi.service('admin::user').sanitizeUser(args[0].user),
+  });
+
+  const eventMap = {
+    'entry.create': defaultGetPayload,
```

Comment:
What's the advantage of storing a function reference here vs invoking the function and storing its value. Especially if we can pass the payload directly instead of `...args`.

Reply:
I think to call the function directly you'd need several if/else statements or a switch case. I find the object a bit more concise. Also I think it's easier to extend an object via a provider system

Reply:
But as I [commented](https://github.com/strapi/strapi/pull/15179#discussion_r1050616327) above, if we don't care about having a custom payload for an event, then we don't need the object